src/pages/Post/hooks/useUpdatePost.ts

```diff
@@ -0,0 +1,53 @@
+import { gql } from "graphql-request";
+import { useMutation } from "react-query";
+
+import {
+    UpdatePostByIdMutationMutation,
+    UpdatePostByIdMutationMutationVariables,
+} from "./../../../generated/graphql";
+
+import useQueryClient from "src/hooks/useQueryClient";
+
+const useUpdatePost = (refetchPosts: () => void, successUpdate: () => void) => {
```

Comment:
I think you should pass in one successUpdate and then do whatever you want outside this function